rules/naming/src/Rector/Assign/RenameVariableToMatchMethodCallReturnTypeRector.php

```diff
@@ -138,28 +139,28 @@ public function getRunner(): Runner
      */
     public function getNodeTypes(): array
     {
-        return [Assign::class];
+        return [Assign::class, Foreach_::class];
```

Comment:
Mixing these 2 types together will create mixed complexity. Foreach and get methods have completely different behavior. 

- One dummy is assign, no behavior happens.
- Foreach changes getter to value and expects array/iterable etc.

Please separate it to 2 rules